lib/client/player_test.go

```diff
@@ -136,6 +136,101 @@ func TestPlayPause(t *testing.T) {
 	require.True(t, p.Stopped())
 }
 
+func TestEndPlaybackWhilePlaying(t *testing.T) {
+	c := clockwork.NewFakeClock()
+
+	// in this test, we let the player play 2 of the 3 events,
+	// then pause it and verify the pause state before resuming
+	// playback for the final event.
+	events := printEvents(100, 200, 300)
+	var stream []byte // intentionally empty, we dont care about stream contents here
+	p := newSessionPlayer(events, stream, testTerm(t))
+	p.clock = c
+
+	p.Play()
+
+	// wait for player to see the first event and apply the delay
+	c.BlockUntil(1)
+
+	// advance the clock:
+	// at this point, the player will write the first event
+	c.Advance(100 * time.Millisecond)
+
+	// wait for the player to sleep on the 2nd event
+	c.BlockUntil(1)
+
+	// end playback
+	p.EndPlayback()
+
+	// advance the clock again:
+	// the player will write the second event and
+	// then realize that it's been asked to end playback
+	c.Advance(100 * time.Millisecond)
+
+	// check that stopC was written to
+	select {
+	case <-time.After(2 * time.Second):
+		panic("timed out waiting for player to complete")
+	case <-p.stopC:
+		require.True(t, p.Stopped())
```

Comment:
nit: add a description, the default error message is not that useful for require.True/False.

```suggestion
		require.True(t, p.Stopped(), "Stopped returned an unexpected value")
```